x-pack/plugins/index_management/public/application/components/component_templates/component_template_details/component_template_details.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiCallOut,
+} from '@elastic/eui';
+
+import { SectionLoading, TabSettings, TabAliases, TabMappings } from '../shared_imports';
+import { useComponentTemplatesContext } from '../component_templates_context';
+import { TabSummary } from './tab_summary';
+import { ComponentTemplateTabs, TabType, Tab } from './tabs';
+import { ManageButton, ManageAction } from './manage_button';
+
+interface Props {
+  componentTemplateName: string;
+  onClose: () => void;
+  showFooter?: boolean;
+  actions?: ManageAction[];
+}
+
+export const ComponentTemplateDetailsFlyout: React.FunctionComponent<Props> = ({
+  componentTemplateName,
+  onClose,
+  actions,
+}) => {
+  const { api } = useComponentTemplatesContext();
+
+  const { data: componentTemplateDetails, isLoading, error } = api.useLoadComponentTemplate(
+    componentTemplateName
+  );
+
+  const [activeTab, setActiveTab] = useState<Tab>(TabType.Summary);
```

Comment:
nit: personally I prefer to use the power of TS here and not using an enum. By typing `''` inside the parenthesis, only valid options will be listed.